openmdao/solvers/petsc_ksp.py

```diff
@@ -178,15 +189,17 @@ def solve(self, rhs_mat, system, mode):
         options = self.options
         self.mode = mode
 
-        self.ksp.setTolerances(max_it=options['maxiter'],
-                               atol=options['atol'],
-                               rtol=options['rtol'])
-
         unknowns_mat = OrderedDict()
         maxiter = self.options['maxiter']
```

Comment:
since you have a temporary for options that you're using for the atol and rtol lookups, you should also use it for maxiter as well.